src/vs/platform/terminal/common/capabilities/shellEnvDetectionCapability.ts

```diff
@@ -36,4 +36,25 @@ export class ShellEnvDetectionCapability extends Disposable implements IShellEnv
 		// Convert to event and fire event
 		this._onDidChangeEnv.fire(this._env);
 	}
+
+	startSingleEnvironmentVar(isTrusted: boolean): void {
+		if (!isTrusted) {
+			return;
+		}
+		this._env.clear();
+	}
+	setSingleEnvironmentVar(key: string, value: string | undefined, isTrusted: boolean): void {
+		if (!isTrusted) {
+			return;
+		}
+		if (key !== undefined && value !== undefined) {
+			this._env.set(key, value);
+		}
+	}
+	endSingleEnvironmentVar(isTrusted: boolean): void {
+		if (!isTrusted) {
+			return;
+		}
+		this._onDidChangeEnv.fire(this._env);
+	}
```

Comment:
Just in case this is accessed between the start and end calls, let's make it transactional and store the `_pendingEnv` in a property. That could be instantiated in `start` and then set `_env = _pendingEnv` add `_pendingEnv = undefined` in `end`

Reply:
This is great point :)

Would this be fine? https://github.com/microsoft/vscode/pull/237602/commits/9032f1646b0b3f8ea9b092e5ad087b21590e0c70

Reply:
More updates here and wrote few lines for the doc: https://github.com/microsoft/vscode/pull/237602/commits/f4c32ebec5cc90d562ce3e76776250554a053f54

Reply:
See new review